### PR TITLE
provider/openstack: LoadBalancer Security Groups

### LINKED ISSUE #11066: Terraform Openstack attach LB to a port or security group to a LB
So, Im trying to use terraform to assign a security group to a loadbalancer, but there doesnt seem to be an option to allow me to set the security group on a openstack_lb_loadbalancer_v2 resource nor does there seem to be an option to assign the security group to a openstack_networking_port_v2 resource and then attach a LB to the networkPort.

**Terraform Version**
Terraform v0.8.2

**Affected Resource(s)**
openstack_lb_loadbalancer_v2 
openstack_networking_port_v2

```
resource "openstack_networking_network_v2" "test_network" {
  	name = "${var.environment_name}-network"
  	admin_state_up = "true"
}
resource "openstack_networking_port_v2" "lb_port" {
  	name = "test_lb_port"
  	network_id = "${openstack_networking_network_v2.test_network.id}"
  	security_group_ids = [
  		"${openstack_compute_secgroup_v2.test.id}"
  	]
  	admin_state_up = "true"
}
resource "openstack_lb_loadbalancer_v2" "test_lb" {
  	name = "${var.environment_name}-test-lb"
  	vip_subnet_id = "${openstack_networking_subnet_v2.test_subnet.id}"
}

resource "openstack_networking_floatingip_v2" "test_lb_vip" {
    pool = "${var.floatingip_pool}"
    port_id = "${openstack_lb_loadbalancer_v2.test_lb.vip_port_id}"
    depends_on = [
        "openstack_lb_loadbalancer_v2.test_lb",
    ]
}

resource "openstack_lb_listener_v2" "test_lb_listener" {
  	protocol = "HTTPS"
  	protocol_port = 443
  	loadbalancer_id =  "${openstack_lb_loadbalancer_v2.test_lb.id}"
}

resource "openstack_lb_pool_v2" "test_lb_pool" {
	name = "${var.environment_name}-test-pool"
  	protocol = "HTTPS"
  	lb_method = "ROUND_ROBIN"
  	listener_id =  "${openstack_lb_listener_v2.test_lb_listener.id}"
}

resource "openstack_lb_member_v2" "test_lb_members" {
	name = "${var.environment_name}-test-member"
  	count = "${var.number_of_test}"
  	pool_id = "${openstack_lb_pool_v2.test_lb_pool.id}"
  	subnet_id = "${openstack_networking_subnet_v2.test_subnet.id}"
  	address = "${element(openstack_compute_instance_v2.test.*.network.0.fixed_ip_v4, count.index)}"
  	protocol_port = 443
  	
  	depends_on = [
        "openstack_lb_loadbalancer_v2.test_lb",
    ]
}
```

### PR DESCRIPTION
**Pull request by jtopjian**
This commit adds the ability to specify security groups on a loadbalancer
resource.

Fixes #11066

### CODE REVIEW

File: builtin/providers/openstack/resource_openstack_lb_loadbalancer_v2.go [hunk 1]
```diff
@@ -126,6 +134,25 @@ func resourceLoadBalancerV2Create(d *schema.ResourceData, meta interface{}) erro
 		return err
 	}
 
+	// Once the loadbalancer has been created, apply any requested security groups
+	// to the port that was created behind the scenes.
+	if lb.VipPortID != "" {
```
Comment (stack72) [inline builtin/providers/openstack/resource_openstack_lb_loadbalancer_v2.go]: Is it worth letting the Create func call the Update func here? That way you wouldn't need to duplicate the code too much

Will that introduce any more risk?

P.

Comment (jtopjian) [inline builtin/providers/openstack/resource_openstack_lb_loadbalancer_v2.go]: That's an interesting idea! I'll try it out.

**Comment by jtopjian** [comment]
@stack72 I wasn't able to successfully call the Update function, but I was able to break the logic out into a dedicated function.

**Comment by stack72** [comment]
LGTM! :)

**Pull request merged**